src/Kmd.Logic.Email.Client/EmailCient.cs

```diff
@@ -0,0 +1,124 @@
+    public sealed class EmailCient : IDisposable
+    {
+        private readonly HttpClient httpClient;
+        private readonly EmailOptions options;
+        private readonly ITokenProviderFactory tokenProviderFactory;
+        private InternalClient internalClient;
+        private string bearerToken;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmailCient"/> class.
+        /// </summary>
+        /// <param name="httpClient">The HTTP client to use. The caller is expected to manage this resource and it will not be disposed.</param>
+        /// <param name="tokenProviderFactory">The Logic access token provider factory.</param>
+        /// <param name="options">The required configuration options.</param>
+        public EmailCient(
+            HttpClient httpClient,
+            ITokenProviderFactory tokenProviderFactory,
+            EmailOptions options)
+        {
+            this.httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+            this.tokenProviderFactory = tokenProviderFactory ?? throw new ArgumentNullException(nameof(tokenProviderFactory));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmailCient"/> class using bearer token.
+        /// </summary>
+        /// <param name="httpClient">The HTTP client to use. The caller is expected to manage this resource and it will not be disposed.</param>
+        /// <param name="options">The required configuration options.</param>
+        /// <param name="bearerToken">Required access token to authenticate with File Security module.</param>
+        public EmailCient(
+           HttpClient httpClient,
+           string bearerToken,
+           EmailOptions options)
+        {
+            this.httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+            this.bearerToken = bearerToken ?? throw new ArgumentNullException(nameof(bearerToken));
+        }
+
+        /// <summary>
+        /// Creates MSExchange Configuration.
+        /// </summary>
+        /// <param name="mSExchangeConfigurationRequestDetails">Configuration details to be created.</param>
+        /// <returns>CreateCertificateResponse.</returns>
+        public async Task<MSExchangeConfigurationResponse> CreateMSExchangeConfiguration(MSExchangeConfigurationRequestDetails mSExchangeConfigurationRequestDetails)
+        {
+            var client = this.CreateClient();
+            var request = new MSExchangeConfigurationCreateRequest(
+                mSExchangeConfigurationRequestDetails.FromAddress,
+                mSExchangeConfigurationRequestDetails.ConfigurationName);
+            using var certificateDetailsResponse = await client.CreateEmailConfigurationWithHttpMessagesAsync(
```

Comment:
Use proper variable name.

Reply:
Changed to `configurationDetailsResponse`